src/v/cloud_storage/remote.cc

```diff
@@ -715,6 +730,82 @@ ss::future<upload_result> remote::delete_objects(
   std::vector<cloud_storage_clients::object_key> keys,
   retry_chain_node& parent) {
     ss::gate::holder gh{_gate};
+
+    if (!is_batch_delete_supported()) {
+        vlog(
+          cst_log.info,
+          "Backend {} does not support batch delete, falling back to "
+          "sequential deletes",
+          _cloud_storage_backend);
+
+        auto keys_empty = keys.empty();
```

Comment:
Why not just return here if keys is empty?

Reply:
changed to early return